### PR TITLE
drop support for Python 3.8

### LINKED ISSUE #1098: Drop support for python 3.8
astropy will be dropping support for python 3.8 in March 2023. I'm assuming the jwst package will follow not long after. Given that, it seems that jwql should also shift to >= python 3.9 in that timeframe.

### PR DESCRIPTION
**Pull request by zacharyburnett**
Numpy is imminently dropping support for Python 3.8

### CODE REVIEW

**Comment by bhilbert4** [comment]
Thanks for working on this @zacharyburnett! It's been on our To Do list for a while, but we haven't managed to get to it.

File: jwql/tests/test_pipeline_tools.py [hunk 1]
```diff
@@ -67,7 +67,6 @@ def test_completed_pipeline_steps():
     assert completed_steps == true_completed
 
 
-@pytest.mark.skipif(ON_GITHUB_ACTIONS, reason='Drizzle import issue with python 3.8')
```
Comment (mfixstsci) [inline jwql/tests/test_pipeline_tools.py:70]: @bhilbert4 so this test has issues with drizzle, do we know if this was solved in python versions `> 3.9`

**Review by mfixstsci — APPROVED** [review]

File: jwql/tests/test_pipeline_tools.py (hunk 1 above)
Comment (bhilbert4) [reply jwql/tests/test_pipeline_tools.py:70]: I don't recall what this import issue is/was. I guess we'll see what happens when the test runs. :)

**Comment by zacharyburnett** [comment]
`Python - 3.8` will need to be removed from the `Required Status Checks` in the branch protections after merge:

image

**Comment by zacharyburnett** [comment]
It also looks like running against the Python 3.10 environment fails due to not being able to parse `jwql-3.10` (?)

https://github.com/spacetelescope/jwql/actions/runs/4731959550/jobs/8397513039#step:8:53
```
SpecNotFound: Invalid name 'jwql-3.10', try the format: user/package
```

**Comment by mfixstsci** [comment]
> It also looks like running against the Python 3.10 environment fails due to not being able to parse `jwql-3.10` (?)
> 
> https://github.com/spacetelescope/jwql/actions/runs/4731959550/jobs/8397513039#step:8:53
> 
> ```
> SpecNotFound: Invalid name 'jwql-3.10', try the format: user/package
> ```

@zacharyburnett yeah I am confused because the name in the yml file: https://github.com/spacetelescope/jwql/blob/develop/environment_python_3_10.yml#L18 looks totally fine. So I am confused. I know GHA doesn't like 3.10 but it has to be parsed a string so I wonder if that has anything to do with the parsing.

**Review by mfixstsci — APPROVED** [review]
Thanks again @zacharyburnett

**Pull request merged**